### PR TITLE
contracts: script for making markdown file with all our contract addresses

### PR DESCRIPTION
**Pull request by gigamesh**
Credit to Pacien Boisson @pakocrew

Please review carefully as this will be the source of truth for the near future

### CODE REVIEW

File: packages/contracts/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
   "types": "dist/index",
   "license": "MIT",
   "scripts": {
-    "all": "yarn clean && yarn build && yarn test && yarn lint:fix && yarn lint",
+    "all": "yarn clean && yarn build && yarn test && yarn lint:fix && yarn lint && yarn generate-markdown",
```
Comment (gigamesh) [inline packages/contracts/package.json:16]: I wasn't sure of the best place to run it but figured it should happen here

Comment (snario) [reply packages/contracts/package.json:16]: Probably should be part of the deploy script.

File: packages/contracts/addresses.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
+# Optimism Regenesis Deployments
+## LAYER 2
+
+### Chain IDs:
+- Mainnet: 10
+- Kovan: 69
```
Comment (tynes) [inline packages/contracts/addresses.md:6]: Do you mind adding goerli as well? Its 420

File: packages/contracts/package.json (hunk 1 above)
Comment (gigamesh) [reply packages/contracts/package.json]: moved 👍

**Review by snario — CHANGES_REQUESTED** [review]
Instead of a column named **Open** could we just hyperlink the address of the contract to make the table width smaller?

**Comment by tynes** [comment]
> Instead of a column named Open could we just hyperlink the address of the contract to make the table width smaller?

I'd support this

File: packages/contracts/scripts/generate-markdown.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+## LAYER 1\n\n`;
+
+  for (let deployment of deployments) {
+    md += `## ${deployment.toUpperCase()}\n\n`;
+
+    const chainId = Number(
+      fs.readFileSync(`./deployments/${deployment}/.chainId`)
+    );
+    const network = networks[chainId];
```
Comment (smartcontracts) [inline packages/contracts/scripts/generate-markdown.js:63]: Could we add some sort of per-network annotations that can be injected into the markdown? E.g., `kovan-v1` is outdated and will no longer be in use. We'll want to stop showing those addresses to people once we stop using the contracts. We probably want to add some sort of text to that section to make it clear that those contracts are deprecated. Also probably want to add the option to hide a deployment completely.

**Comment by smartcontracts** [comment]
Also requesting that we add commit credits for @pakocrew

**Comment by gakonst** [comment]
I rebase and `git commit --amend --author="Pacien Boisson <pakokrew@gmail.com" --no-edit` the first commit to credit @pakokrew

File: packages/contracts/scripts/generate-markdown.js (hunk 3 above)
Comment (gakonst) [reply packages/contracts/scripts/generate-markdown.js:63]: Let's do this in a separate PR, if you have a specific idea could you please open an issue for it?

**Pull request merged**